### PR TITLE
Adds saveLastRetiredFederationP2SHScript test

### PR DESCRIPTION
**Pull request by jeremy-then**
Adds saveLastRetiredFederationP2SHScript test

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 1]
```diff
@@ -998,6 +999,31 @@ void saveNextFederationCreationBlockHeight_whenHeightIsNotNullAndRSKIP186IsActiv
 
     }
 
+    @Test
+    void saveLastRetiredFederationP2SHScript_postIris_getsValueFromStorage() {
```
Comment (nathanieliov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:1003]:
```suggestion
    void saveLastRetiredFederationP2SHScript_whenIrisIsActive_getsValueFromStorage() {
```

**Review by nathanieliov — APPROVED** [review]
Just a small suggestion

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 1 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:945]: Already discussed in the poll.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 2]
```diff
@@ -940,6 +941,31 @@ void saveNextFederationCreationBlockHeight_postIrisAndHeightIsNotNull_shouldSave
...
+        ActivationConfig.ForBlock irisActivations = ActivationConfigsForTest.iris300().forBlock(0L);
+
+        Script expectedScript = new P2shErpFederationBuilder().build().getP2SHScript();
+
+        StorageAccessor storageAccessor = new InMemoryStorage();
+        FederationStorageProvider federationStorageProvider = new FederationStorageProviderImpl(storageAccessor);
+
+        // Act
+
+        federationStorageProvider.setLastRetiredFederationP2SHScript(expectedScript);
```
Comment (wilmerrootstock) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:958]: This line doesn't belong to `Act` part. WDYT if we move it to the `Arrange` part?

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java [hunk 3]
```diff
@@ -940,6 +941,31 @@ void saveNextFederationCreationBlockHeight_postIrisAndHeightIsNotNull_shouldSave
...
     }
 
+    @Test
+    void saveLastRetiredFederationP2SHScript_postIris_getsValueFromStorage() {
+
+        // Arrange
+
+        ActivationConfig.ForBlock irisActivations = ActivationConfigsForTest.iris300().forBlock(0L);
+
+        Script expectedScript = new P2shErpFederationBuilder().build().getP2SHScript();
```
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java:951]:
```suggestion
        Script expectedScript = new P2shErpFederationBuilder().build().getDefaultP2SHScript();
```
Let's use that one that is the one actually stored

Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java]: Updated.

File: rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java (hunk 2 above)
Comment (jeremy-then) [reply rskj-core/src/test/java/co/rsk/peg/federation/FederationStorageProviderImplTests.java]: Updated.

**Review by wilmerrootstock — APPROVED** [review]
LGTM!

**Review by marcos-iov — APPROVED** [review]

**Pull request merged**